### PR TITLE
Functional tso command test cases currently can not be run concurrently#880

### LINKED ISSUE #880: [Enabler] [zos_tso_command] Functional test cases currently can not be run concurrently
### Is there an existing issue for this?

- [X] There are no existing issues.

### Ansible module

_No response_

### Enabler description

The zos_tso_command test cases require sequential execution, thus far these are the ones I have found. These need to be re-written so that they can operate individually without a dependency on one another. I noticed this when I tried to run the load balancer and tests cases failed.

### PR DESCRIPTION
**Pull request by AndreMarcel99**
##### SUMMARY
The change substitute a set of test that depends of a sequential run and summarized in one large test with some of the options tested before and remove hardcoded verifications.

Fixes #880 

##### ISSUE TYPE
- Bugfix Pull Request

##### ADDITIONAL INFORMATION
Take logic of 8 tests and validation of each one, that required a sequential run  to just one test in the right sequence.
Captura de pantalla 2023-07-17 a la(s) 12 50 07

`test_zos_tso_command_allocate_listing_delete`
Captura de pantalla 2023-07-17 a la(s) 12 50 17

### CODE REVIEW

**Review by richp405 — APPROVED** [review]
Pretty slick merger of tests!

File: tests/functional/modules/test_zos_tso_command_func.py [hunk 1]
```diff
@@ -146,7 +114,8 @@ def test_zos_tso_command_invalid_command(ansible_zos_module):
     hosts = ansible_zos_module
     results = hosts.all.zos_tso_command(commands=["xxxxxx"])
     for result in results.contacted.values():
-        assert result.get("output")[0].get("rc") == 255
+        for item in result.get("output"):
+            assert item.get("rc") == 255
         assert result.get("changed") is False
```
Comment (ketankelkar) [inline tests/functional/modules/test_zos_tso_command_func.py:121]: I know you didn't touch this next test `test_zos_tso_command_multiple_commands` but I was wondering if it's possible to verify that 2 commands were run. Currently it looks like the test is only checking that the `rc` is `0`. I'd like to see a check for the correct number of `rc`s coming back (from [docs](https://ibm.github.io/z_ansible_collections_doc/ibm_zos_core/docs/source/modules/zos_tso_command.html#return-values), it appears that for each command there should be a return structure).

File: tests/functional/modules/test_zos_tso_command_func.py [hunk 2]
```diff
@@ -45,89 +47,53 @@ def test_zos_tso_command_long_command_128_chars(ansible_zos_module):
...
-# the dataset has be deleted.
-def test_zos_tso_command_valid_command_failed_as_has_been_deleted(ansible_zos_module):
-    hosts = ansible_zos_module
-    results = hosts.all.zos_tso_command(commands=["delete 'imstestl.ims1.temp.ps'"])
+    # Validate already was remove
+    results = hosts.all.zos_tso_command(commands=["delete '{0}'".format(DEFAULT_TEMP_DATASET)])
     for result in results.contacted.values():
-        assert result.get("output")[0].get("rc") == 8
+        for item in result.get("output"):
+            assert item.get("rc") == 8
```
Comment (ketankelkar) [inline tests/functional/modules/test_zos_tso_command_func.py:96]: A little nit-picking here, I think it would be beneficial to add a comment explaining why the expected rc is 8. Maybe something like:

"# Expect the tso_command to fail here because the previous command will have already deleted the data set"

File: tests/functional/modules/test_zos_tso_command_func.py [hunk 3]
```diff
@@ -45,89 +47,53 @@ def test_zos_tso_command_long_command_128_chars(ansible_zos_module):
...
+            assert item.get("rc") == 0
         assert result.get("changed") is True
-
-
-# The failure path test
-# the dataset has be deleted.
-def test_zos_tso_command_valid_command_failed_as_has_been_deleted(ansible_zos_module):
-    hosts = ansible_zos_module
-    results = hosts.all.zos_tso_command(commands=["delete 'imstestl.ims1.temp.ps'"])
+    # Validate already was remove
```
Comment (ketankelkar) [inline tests/functional/modules/test_zos_tso_command_func.py:92]: consider rewording this comment to:
"Validate data set was removed by previous call"

File: tests/functional/modules/test_zos_tso_command_func.py [hunk 4]
```diff
@@ -45,89 +47,53 @@ def test_zos_tso_command_long_command_128_chars(ansible_zos_module):
...
+            assert item.get("rc") == 0
         assert result.get("changed") is True
-
-
-# The positive path test
-# tests that single command works as well with alias
-def test_zos_tso_command_auth_command_listds_using_alias(ansible_zos_module):
-    hosts = ansible_zos_module
-    results = hosts.all.zos_tso_command(command=["LISTDS 'imstestl.ims1.temp.ps'"])
+    # Validate listds auth of a dataset with commands
```
Comment (ketankelkar) [inline tests/functional/modules/test_zos_tso_command_func.py:66]: this is outside my scope of knowledge, but I'm not seeing where 'auth' and not-'auth' is invoked in the test case. To be fair, I don't see where/how it was invoked in the tests prior to your changes either. I think this is a good opportunity to add testing for that functionality too

File: tests/functional/modules/test_zos_tso_command_func.py [hunk 5]
```diff
@@ -45,89 +47,53 @@ def test_zos_tso_command_long_command_128_chars(ansible_zos_module):
...
+            assert item.get("rc") == 0
         assert result.get("changed") is True
-
-
-# The positive path test
-# tests that alias "command" works
-def test_zos_tso_single_command_auth_command_listds_using_alias(ansible_zos_module):
-    hosts = ansible_zos_module
-    results = hosts.all.zos_tso_command(command="LISTDS 'imstestl.ims1.temp.ps'")
+    # Validate listds auth single command
```
Comment (ketankelkar) [inline tests/functional/modules/test_zos_tso_command_func.py:72]: 'single command' is a bit unclear. Maybe reword this to:
"Validate LISTDS using alias param 'command'"

File: changelogs/fragments/880-Functional_tso_command_test_cases_currently_can_not_be_run_concurrently.yml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+bugfix:
```
Comment (ketankelkar) [inline changelogs/fragments/880-Functional_tso_command_test_cases_currently_can_not_be_run_concurrently.yml:1]: i'd recategorize this to 'trivial' which will skip including this fragment in the changelog since these changes are to the tests and not to the module/module_utils, from a user pespective, there's no change.

**Review by ketankelkar — CHANGES_REQUESTED** [review]
Overall changes look great! Just a few minor edits to comments and a couple of requests to expand test coverage. It might be more appropriate to record the requests as a separate issue and implement them later instead of as part of this PR depending on the level of effort (i'm not sure).

File: changelogs/fragments/880-Functional_tso_command_test_cases_currently_can_not_be_run_concurrently.yml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+trivial:
+- zos_tso_command - Test cases were running in a specific sequence to validated.
+  Change summary the sequence into one test case.
```
Comment (ketankelkar) [inline changelogs/fragments/880-Functional_tso_command_test_cases_currently_can_not_be_run_concurrently.yml:3]: Perhaps reword the message to something like:
- zos_tso_command - Test suite was set up to run sequentially such that certain tests relied on prior test cases. The new changes combine those inter-dependent test cases into a single test case so that each individual test case can now be run stand-alone.

File: tests/functional/modules/test_zos_tso_command_func.py (hunk 4 above)
Comment (AndreMarcel99) [reply tests/functional/modules/test_zos_tso_command_func.py]: I already adapt the comment above to say what command is auth and which not

File: tests/functional/modules/test_zos_tso_command_func.py [hunk 8]
```diff
@@ -45,89 +42,54 @@ def test_zos_tso_command_long_command_128_chars(ansible_zos_module):
...
-        assert result.get("output")[0].get("rc") == 0
+        for item in result.get("output"):
+            assert item.get("rc") == 0
         assert result.get("changed") is True
-
-
-# The positive path test
-def test_zos_tso_command_unauth_command_listcat(ansible_zos_module):
-    hosts = ansible_zos_module
+    # Validate LISTCAT comand and an unauth command
```
Comment (ketankelkar) [inline tests/functional/modules/test_zos_tso_command_func.py:73]: missing an 'm' in 'command'

**Review by ketankelkar — CHANGES_REQUESTED** [review]
thanks for making the requested changes, glad we got the auth stuff cleared up offline, and also thanks for adding the extra check for the two-command test. Just suggesting two very tiny edits and this should be good to go. Remember to re-run the pipeline and paste the results before merging

**Comment by ketankelkar** [comment]
clean pipeline run 
image

**Review by ketankelkar — APPROVED** [review]
lgtm!
